libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SalesforceNetworkPlugin.java

```diff
@@ -165,11 +167,13 @@ else if (response.asBytes().length > 0) {
 
                 @Override
                 public void onError(Exception exception) {
-                    callbackContext.error(exception.getMessage());
+                    String jsonString = "{error: \"" + exception.getMessage() + "\"}";
```

Comment:
Constructing the string without JSONObject because this is called when json parsing fails and avoiding another try/catch

Reply:
If the exception message has ", couldn't we end up with an invalid json string?
It would be safer to do 
```
JSONObject errorJson = new JSONObject(); 
errorJson.put("error", exception.getMessage())
```

Reply:
That's what I started with but since put can throw an exception, what do you think should happen if it fails?
```
JSONObject errorJson = new JSONObject();
try {
    errorJson.put("error", e.getMessage());
} catch (JSONException ex) {
                            
}
```

Reply:
Just log it - I don't think it can actually happen since we are putting a string.